wip: Remove Faraday Middleware

bf4 opened the pull request.

minimul commented on lib/qbo_api/connection.rb:
```diff
@@ -1,5 +1,6 @@
 require 'faraday'
-require 'faraday_middleware'
+require 'faraday/response/json'
+require 'qbo_api/raise_http_error'
```
`require 'qbo_api/raise_http_exception'` ?

minimul commented on lib/qbo_api/connection.rb:
```diff
@@ -68,11 +70,7 @@ def response(resp, entity: nil)
-      when /json/ then JSON.parse(body)
-      else body
-      end
+      resp.body
```
Nice!

minimul commented on lib/qbo_api/raise_http_exception.rb:
```diff
@@ -1,32 +1,43 @@
 require 'nokogiri'
 # @private
-module FaradayMiddleware
+module QboApi
```
That name spacing pattern with the gem is:
```rb
class QboApi
   module Name
  ...
....
```

bf4 replied:
🤦

minimul reviewed (changes requested):
Thanks @bf4 for taking this on!

Greyoxide commented:
Are there any updates on this?  I'd love to use this awesome gem but the Faraday mis-match is messing me up.

bf4 commented:
Personally if me I need to run the test suite etc. I just wrote this
quickly to show the way and then forgot about it,  oops

I think there might also be a behavior change around mime type parsing
which may come up.

On Wed, Dec 28, 2022, 2:17 PM Greyoxide ***@***.***> wrote:

> Are there any updates on this? I'd love to use this awesome gem but the
> Faraday mis-match is messing me up.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/minimul/qbo_api/pull/117#issuecomment-1366866691>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABC4QU3Y6BQ47LHWZU2B7TWPSG27ANCNFSM6AAAAAASU322VA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

minimul commented:
It is not hitting any pain points for me at the moment but I'll merge in quality fix and re-release quickly if you all or someone wants to tackle it.

Greyoxide commented:
I'd love to help but I know so little about the gem or the faraday_middleware gem, I'm worried I'd break something.  If someone wanted to outline specific areas to focus on and what needed to happen, I'd be willing to take a crack at it.

bf4 commented:
The change needed is basically this pr.  I'm not around to test if the
changed are good or not and ci didn't run.  But I've replaced all my other
faraday json usages in my app which used faraday_middleware with a change
like in the pr.

On Thu, Dec 29, 2022, 1:21 PM Greyoxide ***@***.***> wrote:

> I'd love to help but I know so little about the gem or the
> faraday_middleware gem, I'm worried I'd break something. If someone wanted
> to outline specific areas to focus on and what needed to happen, I'd be
> willing to take a crack at it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/minimul/qbo_api/pull/117#issuecomment-1367506777>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABC4QURJ7KILTY4LS5FJ33WPXJC7ANCNFSM6AAAAAASU322VA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

bf4 commented:
Superseded by https://github.com/minimul/qbo_api/pull/118

bf4 commented on qbo_api.gemspec:
```diff
@@ -25,8 +25,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'amazing_print'
-  spec.add_runtime_dependency 'faraday', '~> 1.10.0'
-  spec.add_runtime_dependency 'faraday_middleware'
+  spec.add_runtime_dependency 'faraday', '>= 1.10.0'
```
intention here is that gem can, for now, remain compatible with both 1.x and 2x. per my observations
